prow/cmd/image-builder/main.go

```diff
@@ -0,0 +1,218 @@
+		if strings.HasPrefix(kanikoArg, "--cache=") || strings.HasPrefix(kanikoArg, "--cache-repo=") {
+			return fmt.Errorf("please use --cache-registry option to enable/disable cache")
+		}
+		if strings.HasPrefix(kanikoArg, "--destination=") || strings.HasPrefix(kanikoArg, "--target=") {
+			return fmt.Errorf("please use --registry, --target and --add-[xyz]-tag options to define targets and destinations")
+		}
+		if strings.HasPrefix(kanikoArg, "--dockerfile=") {
+			return fmt.Errorf("please use --dockerfile option to define the path to the dockerfile")
+		}
+	}
+	return nil
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.dockerConfigSecret, "docker-config-secret", "", "secret which includes docker config.json file")
+	fs.StringVar(&o.dockerfile, "dockerfile", "Dockerfile", "path to dockerfile to be built")
+	fs.Var(&o.targets, "target", "target of dockerfile to be built")
+	fs.Var(&o.kanikoArgs, "kaniko-arg", "kaniko-arg for the build")
+	fs.StringVar(&o.registry, "registry", "", "container registry where build artifacts are being pushed. Cache is disabled for empty value")
+	fs.StringVar(&o.cacheRegistry, "cache-registry", "", "container registry where cache artifacts are being pushed")
+	fs.StringVar(&o.kanikoImage, "kaniko-image", "gcr.io/kaniko-project/executor:v1.8.0", "kaniko image for kaniko build")
+	fs.BoolVar(&o.addVersionTag, "add-version-tag", false, "Add label from VERSION file of git root directory to image tags")
+	fs.BoolVar(&o.addVersionSHATag, "add-version-sha-tag", false, "Add label from VERSION file of git root directory plus SHA from git HEAD to image tags")
+	fs.BoolVar(&o.addDateSHATag, "add-date-sha-tag", false, "Using vYYYYMMDD-<rev short> scheme which is compatible to autobumper")
+	fs.Var(&o.addFixedTags, "add-fixed-tag", "Add a fixed tag to images")
+
+	fs.StringVar(&o.logLevel, "log-level", "info", fmt.Sprintf("Log level is one of %v.", logrus.AllLevels))
+
+	err := fs.Parse(os.Args[1:])
+	if err != nil {
+		logrus.Fatalf("Unable to parse command line flags: %v", err)
+	}
+
+	jobSpec, err := downwardapi.ResolveSpecFromEnv()
+	if err != nil {
+		logrus.Fatalf("Unable to resolve prow job spec: %v", err)
+	}
+
+	if jobSpec.Refs != nil {
+		o.org = jobSpec.Refs.Org
+		o.repo = jobSpec.Refs.Repo
+		o.baseSHA = jobSpec.Refs.BaseSHA
+	} else {
+		logrus.Fatal("Unable to find a valid git ref")
+	}
+
+	return o
+}
+
+func getPodNamespace() (string, error) {
+	var namespace string
+
+	versionFile, err := os.Open("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		return "", errors.Wrap(err, "open /var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	}
+	defer versionFile.Close()
+
+	scanner := bufio.NewScanner(versionFile)
+
+	for scanner.Scan() {
+		namespace = scanner.Text()
+		break
+	}
+	if scanner.Err() != nil {
+		return "", errors.Wrap(err, "scan namespace file")
+	}
```

Comment:
This wraps the wrong error
```suggestion
	if err := scanner.Err(); err != nil {
		return "", errors.Wrap(err, "scan namespace file")
	}
```